### PR TITLE
Dtspo 1355

### PR DESCRIPTION
**Pull request by darren221980**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-1355

### Change description ###

Updated sbox config to create / use additional app gw and changed the private ip from static to dynamic.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: modules/app-gateway-module-backend/main.tf [hunk 1]
```diff
@@ -58,8 +58,7 @@ resource "azurerm_application_gateway" "ag" {
   frontend_ip_configuration {
     name                          = "appGwPrivateFrontendIp"
     subnet_id                     = data.azurerm_subnet.app_gw.id
-    private_ip_address            = var.private_ip_address
-    private_ip_address_allocation = "Static"
+    private_ip_address_allocation = "Dynamic"
```
Comment (timja) [inline modules/app-gateway-module-backend/main.tf:61]: this is a bit worrying doing this in a zero downtime way? wouldn't it be safer to allocate the IP address in each agw block in the yaml

Comment (darren221980) [reply modules/app-gateway-module-backend/main.tf]: This has been set back to Static.

File: modules/app-gateway-module-backend/config.tf [hunk 2]
```diff
@@ -41,7 +41,8 @@ data "azurerm_subnet" "app_gw" {
 }
 
 resource "azurerm_public_ip" "app_gw" {
-  name                = "aks-appgw-${var.env}-pip"
+  count               = length(var.private_ip_address)
```
Comment (thomast1906) [inline modules/app-gateway-module-backend/config.tf:44]: Can you remove changes to this module? They have been applied to the standalone module already.
In another PR we can look at removing this module from the folder

Comment (darren221980) [reply modules/app-gateway-module-backend/config.tf]: These changes have been removed.

File: components/cftapps_cluster_lb_backend/main.tf [hunk 3]
```diff
@@ -13,10 +16,16 @@ module "app-gw" {
...
-  backend_pool_ip_addresses = var.cft_apps_cluster_ips
-  subscription              = var.subscription
-  vault_name                = var.certificate_key_vault_name
-  oms_env                   = var.oms_env
+  location                   = var.location
+  private_ip_address         = var.app_gw_private_ip_address
+  backend_pool_ip_addresses  = var.cft_apps_cluster_ips
+  subscription               = var.subscription
+  vault_name                 = var.certificate_key_vault_name
+  oms_env                    = var.oms_env
```
Comment (thomast1906) [inline components/cftapps_cluster_lb_backend/main.tf:24]: Is oms_env needed?

File: components/cftapps_cluster_lb_backend/main.tf [hunk 4]
```diff
@@ -13,10 +16,16 @@ module "app-gw" {
...
-  private_ip_address        = var.app_gw_private_ip_address
-  backend_pool_ip_addresses = var.cft_apps_cluster_ips
-  subscription              = var.subscription
-  vault_name                = var.certificate_key_vault_name
-  oms_env                   = var.oms_env
+  location                   = var.location
+  private_ip_address         = var.app_gw_private_ip_address
+  backend_pool_ip_addresses  = var.cft_apps_cluster_ips
+  subscription               = var.subscription
+  vault_name                 = var.certificate_key_vault_name
```
Comment (thomast1906) [inline components/cftapps_cluster_lb_backend/main.tf:23]: is vault_name needed?

File: components/cftapps_cluster_lb_backend/main.tf [hunk 5]
```diff
@@ -13,10 +16,15 @@ module "app-gw" {
...
-  location                  = var.location
-  private_ip_address        = var.app_gw_private_ip_address
-  backend_pool_ip_addresses = var.cft_apps_cluster_ips
-  subscription              = var.subscription
-  vault_name                = var.certificate_key_vault_name
-  oms_env                   = var.oms_env
+  location                   = var.location
+  private_ip_address         = var.app_gw_private_ip_address
+  backend_pool_ip_addresses  = var.cft_apps_cluster_ips
+  subscription               = var.subscription
```
Comment (thomast1906) [inline components/cftapps_cluster_lb_backend/main.tf:22]: is subscription needed?

**Review by thomast1906 — CHANGES_REQUESTED** [review]

File: components/cftapps_cluster_lb_backend/main.tf (hunk 3 above)
Comment (darren221980) [reply components/cftapps_cluster_lb_backend/main.tf]: oms_env removed

File: components/cftapps_cluster_lb_backend/main.tf [hunk 6]
```diff
@@ -13,10 +16,16 @@ module "app-gw" {
...
```
[the remark this answers is not in the document]
Comment (darren221980) [reply components/cftapps_cluster_lb_backend/main.tf]: this has been removed

File: components/cftapps_cluster_lb_backend/main.tf (hunk 5 above)
Comment (darren221980) [reply components/cftapps_cluster_lb_backend/main.tf]: now removed

**Review by thomast1906 — APPROVED** [review]

File: components/cftapps_cluster_lb_backend/main.tf [hunk 7]
```diff
@@ -1,22 +1,28 @@
-locals {
-  env = (var.env == "aat") ? "stg" : "${(var.env == "perftest") ? "test" : "${var.env}"}"
+data "azurerm_log_analytics_workspace" "log_analytics" {
```
Comment (timja) [inline components/cftapps_cluster_lb_backend/main.tf:1]: this should use the module for looking it up:

https://github.com/hmcts/terraform-module-log-analytics-workspace-id/

File: environments/sbox/backend_lb_config.yaml [hunk 8]
```diff
@@ -54,6 +54,101 @@ gateways:
       - product: plum
         component: frontend
 
+    # SIDAM 
+      - product: idam
```
Comment (timja) [inline environments/sbox/backend_lb_config.yaml:58]: I'm guessing these were used for testing the module and all should be removed now?

Comment (darren221980) [reply environments/sbox/backend_lb_config.yaml:58]: they are currently on both gateways but after a successful merge and deploy we will then do another change to split each of the names (SIDAM, Plum Frontend, etc) equally across both of the gateways for the sbox env.

Comment (thomast1906) [reply environments/sbox/backend_lb_config.yaml:58]: @darren221980 lets have a chat about how we will look at the deployment of additional application gateway.

File: components/cftapps_cluster_lb_backend/main.tf (hunk 7 above)
Comment (darren221980) [reply components/cftapps_cluster_lb_backend/main.tf]: Updated and re-ran here - https://dev.azure.com/hmcts/CNP/_build/results?buildId=124354&view=results

**Review by timja — APPROVED** [review]

**Pull request merged**